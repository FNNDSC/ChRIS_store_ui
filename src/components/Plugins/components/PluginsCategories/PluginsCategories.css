.plugins-categories {
  /* display */
  padding: 0 1em 0;
  
  /* flex */
  flex: 1;
}

.plugins-categories-header {
  /* color */
  color: #4d5258;
  color: var(--pf-black-700);

  /* display */
  margin-top: 22px;
  padding-bottom: 0.25em;
  margin-bottom: 0.5em;

  /* font */
  font-size: 2em;
  font-weight: 600;
  
  /* border */
  border-bottom-color: #4d5258;
  border-bottom: 3px solid var(--pf-black-700);
}

.plugins-category {
  /* color */
  color: #0088ce;
  color: var(--pf-blue-400);

  /* display */
  display: flex;
  padding-bottom: 0.5em;
  
  /* font */
  font-size: 1.4em;
  font-weight: 600;
  line-height: 1.2;

  /* other */
  cursor: pointer;
}

.plugins-category-name { flex: 1; }

.plugins-category-length { padding-left: 1em; }
