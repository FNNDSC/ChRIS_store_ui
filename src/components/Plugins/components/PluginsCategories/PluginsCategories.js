import React from 'react';
import PropTypes from 'prop-types';

import './PluginsCategories.css';

const PluginsCategories = props => (
  <div className="plugins-categories">
    <div className="plugins-categories-header">Categories</div>
    <div>
      {
        props.categories.map(category => (
          <div key={category.name} className="plugins-category">
            <div className="plugins-category-name">{category.name}</div>
            <div className="plugins-category-length">{category.length}</div>
          </div>
        ))
      }
    </div>
  </div>
);

PluginsCategories.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    length: PropTypes.number.isRequired,
  })).isRequired,
};

export default PluginsCategories;
