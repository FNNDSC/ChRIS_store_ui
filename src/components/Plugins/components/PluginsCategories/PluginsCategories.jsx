import React from 'react';
import PropTypes from 'prop-types';

import './PluginsCategories.css';

const PluginsCategories = ({ categories, selected, onSelect }) => (
  <div className="plugins-categories">
    <div className="plugins-categories-header">Categories</div>
    <div role="menu">
      {categories.map(({ name, length }) => (
        <div
          key={name}
          role="menuitem"
          tabIndex="-1"
          className={`plugins-category ${name === selected ? 'selected' : ''}`}
          onClick={() => onSelect(name)}
          onKeyPress={() => {}}
        >
          <div className="plugins-category-name">{name}</div>
          <div className="plugins-category-length">{length}</div>
        </div>
      ))}

      <div role="menuitem" tabIndex="-1" onClick={() => onSelect(null)} onKeyPress={() => {}}>
        <div style={{ color: '#aaa', cursor: 'pointer' }}>Clear</div>
      </div>
    </div>
  </div>
);

PluginsCategories.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      length: PropTypes.number.isRequired,
    })
  ).isRequired,
  selected: PropTypes.string,
  onSelect: PropTypes.func,
};

export default PluginsCategories;
