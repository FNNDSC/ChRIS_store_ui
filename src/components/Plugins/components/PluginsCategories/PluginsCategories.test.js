import React from 'react';
import { shallow, mount } from 'enzyme';
import PluginsCategories from './PluginsCategories';

describe('PluginsCategories', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<PluginsCategories categories={[]} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render plugins-categories div', () => {
    expect(wrapper.find('div.plugins-categories')).toHaveLength(1);
  });

  it('should render 3 divs inside plugins-categories', () => {
    expect(wrapper.find('div')).toHaveLength(3);
  });

  it('should render plugins-categories-header div', () => {
    expect(wrapper.find('div.plugins-categories-header')).toHaveLength(1);
  });
});

const sampleCategories = [
  {
    name: 'sampleCategory1',
    length: 1,
  },
  {
    name: 'sampleCategory2',
    length: 2,
  },
];

describe('mounted PluginsCategories', () => {
  let wrapper;
  let categories;
  beforeEach(() => {
    wrapper = mount(<PluginsCategories categories={sampleCategories} />);
    categories = Array.from(wrapper.find('div.plugins-category'));
  });

  it('should render the correct number of plugins-category divs', () => {
    expect(categories).toHaveLength(2);
  });

  it('should render name value', () => {
    categories.forEach((category, i) => {
      expect(shallow(category)
        .find('div.plugins-category-name')
        .text())
        .toEqual(`sampleCategory${i + 1}`);
    });
  });

  it('should render length value', () => {
    categories.forEach((category, i) => {
      expect(shallow(category)
        .find('div.plugins-category-length')
        .text())
        .toEqual(`${i + 1}`);
    });
  });
});
