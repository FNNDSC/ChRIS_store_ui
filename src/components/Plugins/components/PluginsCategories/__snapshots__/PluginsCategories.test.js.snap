// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginsCategories should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PluginsCategories
    categories={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="plugins-categories-header"
>
          Categories
</div>,
        <div>
          
</div>,
      ],
      "className": "plugins-categories",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Categories",
          "className": "plugins-categories-header",
        },
        "ref": null,
        "rendered": "Categories",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="plugins-categories-header"
>
            Categories
</div>,
          <div>
            
</div>,
        ],
        "className": "plugins-categories",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Categories",
            "className": "plugins-categories-header",
          },
          "ref": null,
          "rendered": "Categories",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
